ci/engine.go

```diff
@@ -276,28 +271,75 @@ func (e *Engine) Scan(ctx context.Context) (string, error) {
 	})
 	ignoreFileNames, err := ignoreFiles.Entries(ctx)
 	if err != nil {
-		return "", err
+		return err
 	}
 
 	ctr := dag.Container().
-		From("aquasec/trivy:0.50.4").
-		WithMountedFile("/mnt/engine.tar", target.AsTarball()).
+		From("aquasec/trivy:0.52.2").
 		WithMountedDirectory("/mnt/ignores", ignoreFiles).
 		WithMountedCache("/root/.cache/", dag.CacheVolume("trivy-cache"))
 
-	args := []string{
-		"image",
-		"--format=json",
-		"--no-progress",
-		"--exit-code=1",
-		"--vuln-type=os,library",
-		"--severity=CRITICAL,HIGH",
-		"--show-suppressed",
-	}
-	if len(ignoreFileNames) > 0 {
-		args = append(args, "--ignorefile=/mnt/ignores/"+ignoreFileNames[0])
-	}
-	args = append(args, "--input", "/mnt/engine.tar")
+	eg, ctx := errgroup.WithContext(ctx)
+
+	eg.Go(func() error {
+		// scan the source code
+		args := []string{
+			"fs",
+			"--format=json",
+			"--no-progress",
+			"--exit-code=1",
+			"--vuln-type=os,library",
+			"--severity=CRITICAL,HIGH",
+			"--show-suppressed",
+		}
+		if len(ignoreFileNames) > 0 {
+			args = append(args, "--ignorefile=/mnt/ignores/"+ignoreFileNames[0])
+		}
+		args = append(args, "/mnt/src")
+
+		// HACK: filter out npm examples :)
+		src := e.Dagger.Source
+		src = src.WithoutDirectory("sdk/rust/crates/dagger-sdk/examples")
+		src = src.WithoutDirectory("examples/sdk/nodejs")
```

Comment:
@TomChv any ideas about this? :cry: 

I genuinely cannot seem to find a way to get this into a good state right now - it seems like the `ws` dependency is vulnerable - and `websocket-stream` (somewhere in our dep tree) uses a horrendously old version that hasn't got a backport (and of course, looks like `websocket-stream` is abandoned).

IMO I think it's totally reasonable to ignore vulnerabilities in the non-dagger example code, but I wonder if we would want to rework our examples so we don't have this problem in the first place.

Reply:
FYI, this same issue is affecting the `docs` directory.

Reply:
I've expanded this selection to exclude `docs` for now.

Not that those aren't issues, but IMO the `engine scan` pipeline should scan for issues in the engine/sdks/etc - not necessarily the examples and docs, and such.

Reply:
That's a good point! I agree that we should change our example to not use outdated dependencies
@vikram-dagger, could you work on this if you have some time?